src/components/map/Map.js

```diff
@@ -0,0 +1,132 @@
+import React from 'react';
+import L from 'leaflet';
+import { Draw, Control } from 'leaflet-draw';
+
+const MAP_MIN_ZOOM = 2;
+const MAP_INITIAL_ZOOM = 3;
+const BASEMAP_TILE = 'http://server.arcgisonline.com/ArcGIS/rest/services/World_Imagery/MapServer/tile/{z}/{y}/{x}';
+const MAP_CENTER = [51.505, -0.09];
+const BASEMAP_ATTRIBUTION = 'Tiles &copy; Esri &mdash; Source: Esri, i-cubed, USDA, USGS, AEX, GeoEye, Getmapping, Aerogrid, IGN, IGP, UPR-EGP, and the GIS User Community';
+
+const activeLayers = new L.FeatureGroup();
+const drawControlFull = {
+  position: 'topright',
+  draw: {
+      polyline: false,
+      polygon: {
+          allowIntersection: false, // Restricts shapes to simple polygons
+          drawError: {
+              color: '#e1e100', // Color the shape will turn when intersects
+              message: '<strong>Oh snap!<strong> you can\'t draw that!' // Message that will show when intersect
+          },
+          shapeOptions: {
+              color: '#bada55'
+          }
+      },
+      circle: false,
+      rectangle: {
+          shapeOptions: {
+              clickable: false
+          }
+      },
+      marker: false
+  },
+  edit: {
+      featureGroup: activeLayers, //REQUIRED
+      remove: true
+  }
+};
+const drawControlEdit = {
+  position: 'topright',
+  draw: {
+      polyline: false,
+      polygon: false,
+      circle: false,
+      rectangle: false,
+      marker: false
+  },
+  edit: {
+      featureGroup: activeLayers, //REQUIRED
+      remove: true
+  }
+};
+
+class Map extends React.Component {
+
+  componentDidMount() {
+    this.initMap();
+    this.initLayers();
+
+    if (this.props.editable) {
+      this.initDrawing();
+    }
+  }
+
+  componentWillUnmount() {
+    this.remove();
+  }
+
+  onDrawEventComplete(e) {
+    const layer = e.layer;
+    activeLayers.addLayer(layer);
+    if (Object.keys(activeLayers._layers).length) {
```

Comment:
If you're adding a layer to activeLayers before checking if there are layers in activeLayers, will not this always be true? Apart from that I think it's better to use the API method `getLayers()` to check if there is any layers.
Change activeLayers from a global constant to a class property.